libraries/metadata_util.rb

```diff
@@ -0,0 +1,24 @@
+include Chef::Mixin::ShellOut
+
+module MacOS
+  class MetadataUtil
+    attr_reader :status_flags
+
+    def initialize(volume)
+      mdutil_possible_states = { 'Indexing enabled.' => ['on', ''],
+                                 'Indexing disabled.' => ['off', ''],
+                                 'Indexing and searching disabled.' => ['off', '-d'] }
+
+      @status_flags = mdutil_possible_states[volume_current_state(volume)]
+                      .insert(1, volume)
+    end
+
+    def volume_current_state(volume)
+      shell_out("/usr/bin/mdutil -s #{volume.shellescape}")
+        .stdout.split(':')[1].strip
```

Comment:
This is a nitpick, but we do want to be careful of method chaining; as well as taking advantage of `shell_out`'s ability to take any number of arguments for its constructor (doing so _may_ allow use to avoid needing `shellescape`?)

```ruby
def volume_current_state(volume)
  status_output = shell_out('/usr/bin/mdutil', '-s', volume.shellescape).stdout.strip # handle the String first
  status_output.split(':')[1] # and the Array second
end
```

Reply:
I've updated this method to let `shell_out` handle escape characters. Who knew? ;-)